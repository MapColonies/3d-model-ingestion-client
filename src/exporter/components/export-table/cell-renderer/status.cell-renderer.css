.statusPending {
  color: black;
}
.statusIn-Progress {
  color: orange;
}
.statusCompleted {
  color: green;
}
.statusFailed {
  color: red;
}
