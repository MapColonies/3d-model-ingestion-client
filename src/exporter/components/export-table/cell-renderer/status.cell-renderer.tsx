import React from 'react';
import { ICellRendererParams } from 'ag-grid-community';
import { Box } from '@map-colonies/react-components';
import { IExportTaskStatus } from '../../../models/exportTaskStatus';

import './status.cell-renderer.css';

export const StatusRenderer: React.FC<ICellRendererParams> = (
  props
) => {
  const value: string = (props.data as IExportTaskStatus).status;
  
  if (!value) {
    return <></>;//''; // not null!
  }
  return (
    <Box className={`status${value}`}>{value}</Box>
  );

};
