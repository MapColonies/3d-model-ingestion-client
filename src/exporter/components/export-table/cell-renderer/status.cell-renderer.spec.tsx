import React from 'react';
import { shallow } from 'enzyme';
import { ICellRendererParams, Column, RowNode, GridApi, ColumnApi } from 'ag-grid-community';
import { StatusRenderer } from './status.cell-renderer';

/* eslint-disable */
const mockDataBase:ICellRendererParams = {
  value: '',
  valueFormatted: null,
  getValue: () => {},
  setValue: () => {},
  formatValue: () => {},
  data: {status: ''} as any,
  node: new RowNode(),
  colDef: {},
  column: new Column({},null,'status',false),
  $scope: null,
  rowIndex: 1,
  api: new GridApi(),
  columnApi: new ColumnApi(),
  context: null,
  refreshCell: () => {},
  eGridCell: document.createElement('span'),
  eParentOfValue: document.createElement('span'),
  addRenderedRowListener: () => {},
};
/* eslint-enable */

describe('AgGrid StatusRenderer component', () => {
  it('renders correctly', () => {
    const mockData = {
      ...mockDataBase
    };

    const wrapper = shallow(
      <StatusRenderer {...mockData} />
    );

    expect(wrapper).toMatchSnapshot();
  });

});
