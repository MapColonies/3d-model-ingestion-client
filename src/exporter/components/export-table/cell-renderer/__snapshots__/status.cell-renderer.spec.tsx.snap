// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgGrid StatusRenderer component renders correctly 1`] = `<Fragment />`;
